harness/determined/exec/harness.py

```diff
@@ -39,6 +46,9 @@ def main(train_entrypoint: str) -> int:
     # TFKerasTrialController or EstimatorTrialController to add that functionality, so for now we
     # continue with the legacy strategy.
 
+    if pytorch and issubclass(trial_class, pytorch.PyTorchTrial):
```

Comment:
This is going to be an attribute error if pytorch failed to import.

Reply:
side note, this if statement should be next to the trial class, not underneath the long ~rant~ comment about EnvContext.